lib/Interpreter/Interpreter.cpp

```diff
@@ -356,95 +365,97 @@ namespace cling {
     // itself? One could use a macro (simillar to __dso_handle) to block
     // assignemnt and get around the mangling issue.
     const char* Linkage = LangOpts.CPlusPlus ? "extern \"C\"" : "";
-    if (!NoRuntime && !SyntaxOnly) {
+    if (!NoRuntime) {
       if (LangOpts.CPlusPlus) {
         Strm << "#include \"cling/Interpreter/RuntimeUniverse.h\"\n"
                 "namespace cling { class Interpreter; namespace runtime { "
-                "Interpreter* gCling=(Interpreter*)" << thisP << "; }}\n";
+                "Interpreter* gCling";
+        AssignIfValue(Strm, ThisP, "Interpreter*", " }}");
```

Comment:
Fine, I can investigate why setting `gCling` would fail in syntax-only (I don't see it yet), and I can do that after merging this PR.

Still, could you remove `AssingIfValue` and just spell out `<<`?